docs/source/adrs/0005-inline-cardid-data-into-cardid-before-pushing-to-gov-uk-paas.md

```diff
@@ -0,0 +1,62 @@
+---
+hide_in_navigation: true
+---
+
+# ADR 5: Inline cardid-data into cardid before pushing to GOV.UK PaaS
+
+Date: 2020-01-30
+
+## Status
+
+Accepted
+
+## Context
+
+The [pay-cardid](https://github.com/alphagov/pay-cardid) microservice is
+responsible for retrieving card information for a given card number.
+
+To do this, it requires a number of proprietary .csv files, which are provided
+by various payment service providers. This data is sensitive in the sense
+that it's encumbered by licence restrictions, so we need to take appropriate care
+not to make it public.
+
+The data is roughly 36 MB uncompressed, and changes rarely (a few times a year).
+
+In the current infrastructure, cardid includes the sensitive cardid-data in a git
+submodule which refers to a private repository on GitHub. This is built into the
+pay-cardid docker image, which is then pushed to a private container registry.
+
+We would like to deploy cardid to GOV.UK PaaS using the java buildpack. For
+consistency with the other open source applications, we would like the build
+artefact (a .jar file) to be in a public GitHub release. This means we cannot
+include the sensitive data in the build artefact.
+
+## Decision
+
+When concourse deploys cardid to GOV.UK PaaS it will:
+
+* download the cardid build artefact from github releases
+* clone the pay-cardid repository recursively (which will include the private pay-cardid-data)
+* insert the data from pay-cardid-data into the cardid build artefact
```

Comment:
just to be super clear - this means unzipping the build artefact and updating the jar with the relevant files

Reply:
Yes, that's right. `zip` and `tar` have options to append data to an archive without extracting it, but it's the same idea.